### ISSUE TITLE
[Editor] Scene widget for switching ortho views

### ISSUE THREAD

**Issue by mattbenic** [AtomicGameEngine/AtomicGameEngine#1143]
There are currently shortcuts for different orthographic views in the editor, having a widget in the scene view to toggle them would be good. Related to that would be a visible toggle for the grid.

Probably related to https://github.com/AtomicGameEngine/AtomicGameEngine/issues/779

**Comment by JoshEngebretson** [comment]
I've assigned this to Build 2 (due December 15th), will close #779 as this resolves that general issue

**Comment by mattbenic** [comment]
Does this cover all the features in #779? This seems like a small subset of that task.

**Comment by JoshEngebretson** [comment]
Yes, it covers the issue "Viewport Menu" with the description "See the attached picture for some idea on the features." in that it will add a Viewport Menu with some of the features, specific features should be detailed in subsequent issues or a group issue.

**Comment by mattbenic** [comment]
Ok, when it's implemented, I'll have Jeremey take a look at it and add followup issues if they're needed.

**Comment by JoshEngebretson** [comment]
@JimMarlowe Here's the reference branch I started on:  https://github.com/AtomicGameEngine/AtomicGameEngine/tree/JME-ATOMIC-1143

Let me know if you have any questions.  The main thing is that we can do absolute positioning inside the widget size event.  The scene transform tools and local/world axis toggle should be moved to scene editor view too, the branch has that as well (I think).  Thanks!

**Comment by Alan-FGR** [comment]
I've been thinking... when you're in ortho mode, a fly through cam makes no sense since you're looking at the scene from a point infinitely far away (can't really be 'inside' it without clipping'), so why not also use different controls for the ortho projection? That way it would be much more suitable for 2D editing, for example, WASD could just pan around instead of using Q and E which are not intuitive... Ideally mouse panning would be great too, as well as some kind of intelligent orbiting (I know there's an orbit hotkey already) because other than that the controls don't make much sense for 2D imho.

**Comment by Alan-FGR** [comment]
Maybe for ortho 2.5D like RTSs and such, we could have a mode that pans on X and Z but keeps the camera on the same height... that would surely be useful for editing 2.5D maps and such imo.

**Comment by JimMarlowe** [comment]
hey @JoshEngebretson if you consider this fun work, please continue, there are plenty of fish in this sea.

**Comment by JoshEngebretson** [comment]
Ok, no problem
